contracts/ExoticPositionalMarkets/ExoticPositionalFixedMarket.sol

```diff
@@ -127,30 +130,39 @@ contract ExoticPositionalFixedMarket is Initializable, ProxyOwned, OraclePausabl
         emit NewPositionTaken(msg.sender, _position, fixedTicketPrice);
     }
 
-    function withdraw() external notPaused nonReentrant {
+    function withdraw(address collateral) external notPaused nonReentrant {
         require(withdrawalAllowed, "Not allowed");
         require(canUsersPlacePosition(), "Market resolved");
         require(block.timestamp <= withdrawalPeriod, "Withdrawal expired");
         require(userPosition[msg.sender] > 0, "Not a ticket holder");
         require(msg.sender != marketManager.creatorAddress(address(this)), "Can not withdraw");
-        uint withdrawalFee =
-            fixedTicketPrice.mul(marketManager.withdrawalPercentage()).mul(ONE_PERCENT).div(HUNDRED_PERCENT);
+        uint withdrawalFee = fixedTicketPrice.mul(marketManager.withdrawalPercentage()).mul(ONE_PERCENT).div(
+            HUNDRED_PERCENT
+        );
         totalUsersTakenPositions = totalUsersTakenPositions.sub(1);
         ticketsPerPosition[userPosition[msg.sender]] = ticketsPerPosition[userPosition[msg.sender]].sub(1);
         userPosition[msg.sender] = 0;
-        thalesBonds.transferFromMarket(marketManager.safeBoxAddress(), withdrawalFee.div(2));
-        thalesBonds.transferFromMarket(marketManager.creatorAddress(address(this)), withdrawalFee.div(2));
-        thalesBonds.transferFromMarket(msg.sender, fixedTicketPrice.sub(withdrawalFee));
+        thalesBonds.transferFromMarket(marketManager.safeBoxAddress(), withdrawalFee.div(2), marketManager.paymentToken());
```

Comment:
on withdraw we will send sUSD to safebox and to creator but user can choose the collateral by passing collaterals address